### PR TITLE
Fix edge case in normalizer for keyword args

### LINKED ISSUE #13895: Crash in `Macro.to_string`
### Elixir and Erlang/OTP versions

1.17.3, 1.15.7

### Operating system

any

### Current behavior

I noticed this crash report
```
    ** (MatchError) no match of right hand side value: {:__block__, [line: 19], ["foo"]}
        (elixir 1.17.3) <REDACTED: user-file-path>:1333: Code.Formatter.interpolation_to_algebra_5
        (elixir 1.17.3) <REDACTED: user-file-path>:504: Code.Formatter.quoted_to_algebra_3
        (elixir 1.17.3) <REDACTED: user-file-path>:589: anonymous fn_3 in Code.Formatter.block_args_to_algebra_4
        (elixir 1.17.3) <REDACTED: user-file-path>:2015: Code.Formatter.each_quoted_to_algebra_without_comments_4
        (elixir 1.17.3) <REDACTED: user-file-path>:2001: Code.Formatter.quoted_to_algebra_with_comments_6
        (elixir 1.17.3) <REDACTED: user-file-path>:594: Code.Formatter.block_args_to_algebra_4
        (elixir 1.17.3) <REDACTED: user-file-path>:159: Code.Formatter.to_algebra_2
        (elixir 1.17.3) <REDACTED: user-file-path>:1136: Macro.to_string_1
```
and something older
```
    ** (FunctionClauseError) no function clause matching in anonymous fn_1 in Code.Normalizer.normalize_kw_blocks_4
        (elixir 1.15.7) <REDACTED: user-file-path>:453: anonymous fn({:__cursor__, [closing: [line: 23, column: 160], line: 23, column: 149], []}) in Code.Normalizer.normalize_kw_blocks_4
        (elixir 1.15.7) <REDACTED: user-file-path>:1693: Enum."-map_2-lists^map_1-1-"_2
        (elixir 1.15.7) <REDACTED: user-file-path>:1693: Enum."-map_2-lists^map_1-1-"_2
        (elixir 1.15.7) <REDACTED: user-file-path>:453: Code.Normalizer.normalize_kw_blocks_4
        (elixir 1.15.7) <REDACTED: user-file-path>:1359: Code.quoted_to_algebra_2
        (elixir 1.15.7) <REDACTED: user-file-path>:1186: Macro.to_string_1
```

No idea what AST triggered it

### Expected behavior

No crash

### PR DESCRIPTION
**Pull request by josevalim**
Closes #13895.
Closes #13916.

### CODE REVIEW

**Comment by josevalim** [comment]
CI is failing for other reasons. Please merge if you are happy @sabiwara!

File: lib/elixir/lib/code/normalizer.ex [hunk 1]
```diff
@@ -364,9 +364,13 @@ defmodule Code.Normalizer do
...
-                last_args
+            {:__block__, _meta, [[{{:__block__, key_meta, _}, _} | _] = keyword]} ->
+              if key_meta[:format] == :keyword or block_keyword?(keyword) do
+                [
+                  Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
+                    {{:__block__, [format: :keyword] ++ meta, args}, value}
+                  end)
+                ]
               else
                 [last_arg]
```
Comment (sabiwara) [inline lib/elixir/lib/code/normalizer.ex:375]: Isn't the `Enum.map` only needed in the `or` case?
Something like

```suggestion
              cond do
                key_meta[:format] == :keyword ->
                  keyword
                block_keyword?(keyword) ->
                  [
                    Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
                      {{:__block__, [format: :keyword] ++ meta, args}, value}
                    end)
                  ]
              true ->
                [last_arg]
```

**Review by sabiwara — APPROVED** [review]
Thank you @josevalim  💜
LGTM, just one question

File: lib/elixir/lib/code/normalizer.ex [hunk 2]
```diff
@@ -364,10 +364,19 @@ defmodule Code.Normalizer do
...
+                  [keyword]
+
+                block_keyword?(keyword) ->
+                  [
+                    Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
+                      {{:__block__, [format: :keyword] ++ meta, args}, value}
+                    end)
+                  ]
+
+              true ->
```
Comment (josevalim) [inline lib/elixir/lib/code/normalizer.ex:379]:
```suggestion
                true ->
```

File: lib/elixir/lib/code/normalizer.ex [hunk 3]
```diff
@@ -364,10 +364,19 @@ defmodule Code.Normalizer do
...
+
+                block_keyword?(keyword) ->
+                  [
+                    Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
+                      {{:__block__, [format: :keyword] ++ meta, args}, value}
+                    end)
+                  ]
+
+                true ->
                 [last_arg]
```
Comment (josevalim) [inline lib/elixir/lib/code/normalizer.ex:380]:
```suggestion
                  [last_arg]
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**